<launch>
  <arg name="input_topic"           default="/elevation_mapping/elevation_map_raw"/>
  <arg name="output_topic"          default="/elevation_mapping/elevation_map_filtered"/>
  <arg name="filter_chain_config"   default="$(find grid_map_filters_drs)/config/filter_chain_example.yaml"/>
  <arg name="visualizations_config" default="$(find grid_map_filters_drs)/config/visualizations_example.yaml"/>
  <arg name="launch_prefix"         default=""/>

  <!-- Elevation map filter node. -->
  <node pkg="grid_map_filters_drs" type="grid_map_filter_drs_node" name="elevation_map_filter" output="screen" launch-prefix="$(arg launch_prefix)">
    <!-- Input topic -->
    <param name="input_topic" value="$(arg input_topic)" />
    <!-- Output topic -->
    <param name="output_topic" value="$(arg output_topic)" />

    <!-- Load grid map filter chain configuration -->
    <rosparam command="load" file="$(arg filter_chain_config)" />
  </node>

  <!-- Launch the grid map visualizer. -->
  <node pkg="grid_map_visualization" type="grid_map_visualization" name="grid_map_visualization" output="screen">
    <!-- Input topic -->
    <param name="grid_map_topic" value="$(arg output_topic)" />

    <!-- Load visualziations configuration -->
    <rosparam command="load" file="$(arg visualizations_config)" />
  </node>

</launch>
